superset/daos/base.py

```diff
@@ -48,6 +49,11 @@ class BaseDAO:
     """
     id_column_name = "id"
 
+    def __init_subclass__(cls) -> None:  # pylint: disable=arguments-differ
+        cls.model_cls = get_args(
+            cls.__orig_bases__[0]  # type: ignore  # pylint: disable=no-member
+        )[0]
+
```

Comment:
Wow I didn't know Python has added core support for generic types! Adding a reference here for other reviewers: https://peps.python.org/pep-0560/